test/threads.jl

```diff
@@ -360,6 +360,54 @@ end
     end
 end
 
+@testset "io_thread" begin
+    function io_thread_test()
+        # This test creates a thread that does IO and then blocks the main julia thread
+        # This test hangs if you don't spawn an IO thread.
+        # It hanging or not is technically a race but I haven't seen julia win that race yet.
+        cmd = """
+        Base.Experimental.make_io_thread()
+        function callback()::Cvoid
+            println("Running a command")
+            run(`echo 42`)
+            return
+        end
+        function call_on_thread(callback::Ptr{Nothing})
+            tid = UInt[0]
+            threadwork = @cfunction function(arg::Ptr{Cvoid})
+                current_task().donenotify = Base.ThreadSynchronizer()
+                Base.errormonitor(current_task())
+                println("Calling Julia from thread")
+                ccall(arg, Cvoid, ())
+                nothing
+            end Cvoid (Ptr{Cvoid},)
+            err = @ccall uv_thread_create(tid::Ptr{UInt}, threadwork::Ptr{Cvoid}, callback::Ptr{Cvoid})::Cint
+            err == 0 || Base.uv_error("uv_thread_create", err)
+            gc_state = @ccall jl_gc_safe_enter()::Int8
+            err = @ccall uv_thread_join(tid::Ptr{UInt})::Cint
+            @ccall jl_gc_safe_leave(gc_state::Int8)::Cvoid
+            err == 0 || Base.uv_error("uv_thread_join", err)
+            return
+        end
+        function main()
+            callback_ptr = @cfunction(callback, Cvoid, ())
+            call_on_thread(callback_ptr)
+            println("Done")
+        end
+        main()
+
+        """
+        proc = run(pipeline(`$(Base.julia_cmd()) -e $cmd`), wait=false)
+        sleep(2) # Is there a better way to do this?
```

Comment:
that is too fast
```
threads                                           (1) |         failed at 2024-09-09T17:05:20.339
2024-09-09 13:05:20 EDT	Error During Test at /cache/build/tester-amdci5-15/julialang/julia-master/julia-0fbc8f2e3a/share/julia/test/threads.jl:408
2024-09-09 13:05:20 EDT	  Test threw exception
2024-09-09 13:05:20 EDT	  Expression: io_thread_test()
2024-09-09 13:05:20 EDT	  Process did not exit in time
2024-09-09 13:05:20 EDT	  Stacktrace:
```

Reply:
Something like 10 seconds?

Reply:
https://github.com/JuliaLang/julia/issues/55240#issuecomment-2251870278

> Any test that tries to do timing on CI is just bad. If the system is loaded, it can block for arbitrary amounts of time.